ttnn/cpp/ttnn/operations/data_movement/clone/device/clone_program_factory.cpp

```diff
@@ -0,0 +1,174 @@
+// SPDX-FileCopyrightText: © 2024 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "clone_device_operation.hpp"
+#include "tt_metal/common/work_split.hpp"
+#include "ttnn/deprecated/tt_dnn/op_library/math.hpp"
+
+namespace ttnn::operations::data_movement::clone {
+CloneOperation::ProgramFactory::cached_program_t CloneOperation::ProgramFactory::create(
+    const operation_attributes_t& operation_attributes,
+    const tensor_args_t& tensor_args,
+    tensor_return_value_t& output) {
+    using namespace tt;
+    using namespace tt::tt_metal;
+    using namespace tt::tt_metal::detail;
+
+    const auto& input = tensor_args.input;
+    Program program = Program();
+
+    bool tilized = output.get_layout() == Layout::TILE;
+
+    tt::DataFormat input_cb_data_format = datatype_to_dataformat_converter(input.get_dtype());
+    tt::DataFormat output_cb_data_format = datatype_to_dataformat_converter(output.get_dtype());
+    bool convert_dtype = input_cb_data_format != output_cb_data_format;
+    uint32_t input_unit_size =
+        tilized ? TileSize(input_cb_data_format) : input.get_legacy_shape()[-1] * input.element_size();
```

Comment:
Can you replace all instance of `.get_legacy_shape/.get_shape()` in this PR with the new methods (`get_logical_shape()/get_padded_shape()`) introduced in https://github.com/tenstorrent/tt-metal/pull/13372?

Reply:
I changed `get_legacy_shape` to `get_logical_shape` for `get_shape`.

In `ttnn/cpp/ttnn/operations/data_movement/clone/device/clone_device_operation.cpp`, the following `get_shape`:

```C
CloneOperation::shape_return_value_t CloneOperation::compute_output_shapes(
    const operation_attributes_t& operation_attributes, const tensor_args_t& tensor_args) {
    return tensor_args.input.get_shape();
};
```

will not be changed because `create_device_tensor` requires an argument of type `ttnn::Shape`, which is the output of `compute_output_shapes`.